src/include/storage/write_ahead_log/log_manager.h

```diff
@@ -153,7 +166,7 @@ class LogManager : public common::DedicatedThreadOwner {
   std::vector<BufferedLogWriter> buffers_;
   // The queue containing empty buffers which the serializer thread will use. We use a blocking queue because the
   // serializer thread should block when requesting a new buffer until it receives an empty buffer
-  common::ConcurrentBlockingQueue<BufferedLogWriter *> empty_buffer_queue_;
```

Comment:
@mbutrovich This got pulled out to DBMain so that other components, e.g., replication manager, can return buffers to it.

Reply:
What do we believe the argument for this data structure is? Performance over `malloc`? Back-pressure on the serializer if the disk consumer falls behind?

Reply:
*was, I believe it was backpressure. This is more or less leaving things as they are.